src/Servers/Kestrel/Core/src/Internal/Http/HttpProtocol.FeatureCollection.cs

```diff
@@ -250,5 +251,27 @@ void IHttpRequestLifetimeFeature.Abort()
         }
 
         protected abstract void ApplicationAbort();
+
+        Task IHttpResponseStartFeature.StartAsync(bool flush, CancellationToken cancellationToken)
+        {
+            if (flush)
+            {
+                return FlushAsync(cancellationToken);
```

Comment:
So if flush is set, InitializeResponseAsync(0) never called!? This flush parameter is super confusing and unnecessary.

StartAsync should either no-op or throw (as in pick one of those two options and do that consistently) any time it's called after the response has started.

Reply:
FlushAsync already calls InitializeResponseAsync internally.
https://github.com/aspnet/AspNetCore/blob/e8bc2651204f247b0f0d5ffa784d6b29eca0f0aa/src/Servers/Kestrel/Core/src/Internal/Http/HttpProtocol.cs#L772-L776

Reply:
Cool. That brings us back to the original point which is that you the flush parameter is redundant. Why wouldn't the caller just flush manually using either the pipe or the stream after calling StartAsync if that's what they want?

Reply:
Thinking I agree with @halter73 here, it seems too redundant and confusing, especially if we rename the api to CommitHeaders.